### PR TITLE
added description_for_merchant

### PR DESCRIPTION
**Pull request by AlexKratky**
http://redmine.havelholding.cz/issues/3981

### CODE REVIEW

File: src/Model/SimplePayment.php [hunk 1]
```diff
@@ -235,6 +241,15 @@ public function getDescription()
         return $this->description;
     }
 
+    /**
+     * @return string
```
Comment (Triplkrypl) [inline src/Model/SimplePayment.php]: return type is incorrect, private property can contains `string|null`

File: src/Model/SimplePayment.php [hunk 2]
```diff
@@ -132,6 +137,7 @@ public function __construct($values)
         $this->validTo = new DateTime($data['valid_to']);
         $this->fee = $data['fee'];
         $this->description = $data['description'];
+        $this->description_for_merchant = $data['description_for_merchant'];
```
Comment (Triplkrypl) [inline src/Model/SimplePayment.php]: This is break change, it is somethink like new required parameter. If some one use `SimplePayment` contructor for some reason, without this key his code will crash.
Add some check if `description_for_merchant` key realy exist in input array.

**Review by Triplkrypl — CHANGES_REQUESTED** [review]

**Comment by Triplkrypl** [comment]
New response parameter is not realeaset yet, change target branche in to `v1.x-stage`.

**Comment by AlexKratky** [comment]
> 

but there is not v1.x-stage, only v1.x-spring and this is for summer release https://github.com/ThePay/api-client/branches/all and I am not sure how to create branch to keep checks (one approval etc.)

**Comment by Triplkrypl** [comment]
We have checks for all branches started with **v**, but if we fix break change here, we can release new PHP property without deploying changes on our server. SDK always return null until we kick out new release but merchants developers can implement new interface simultaneously, while we testing new features. This can speed-up integration process.

**Review by Triplkrypl — CHANGES_REQUESTED** [review]
Break change in SimplePayment is not fixed.

**Comment by AlexKratky** [comment]
> Break change in SimplePayment is not fixed.

but if we release it with summer release, there will be that key in response, so no need to check if key exists, otherwise I can add key exist check, but then we can merge it to v1.x

**Comment by Triplkrypl** [comment]
Yes we will always send `description_for_merchant` key in response, but everyone could create `SimplePayment` object in their project for some creative reason, we can not prevent of call the constructor outside of our library.
So new required input key can break some code, i hope only unit tests where instance can be used as dummy mock.
This break changes not affect main interface for developers, but we can not predict how they use our code, i truly recommend check new array key and have new feature without break change as possible.

**Comment by AlexKratky** [comment]
> Yes we will always send `description_for_merchant` key in response, but everyone could create `SimplePayment` object in their project for some creative reason, we can not prevent of call the constructor outside of our library. So new required input key can break some code, i hope only unit tests where instance can be used as dummy mock. This break changes not affect main interface for developers, but we can not predict how they use our code, i truly recommend check new array key and have new feature without break change as possible.

Well now I understand, however it is not documented to use it like that. Every change in input/output of any method could be BC, even after adding check if key is set, someone can have a test, that will check count of key of method toArray() for example

**Review by Triplkrypl — APPROVED** [review]

**Pull request merged**